src/common/CMakeLists.txt

```diff
@@ -49,6 +49,10 @@ if (STACK_TRACE)
   list(APPEND common_sources stack_trace.cpp)
 endif()
 
+if (BACKCOMPAT)
+  list(APPEND common_sources ../compat/glibc_compat.cpp)
```

Comment:
That's not nice to pull things from another directory. Why not have this is common, or have compat have its own cmake file ?

Reply:
I'll probably have to add extra sources for macOS compatibility as well. I'll put the compat directory in common insead, that makes more sense.